pkg/controller/csr/csr_controller.go

```diff
@@ -393,3 +445,17 @@ func (r *reconcileCSR) getPod(ctx context.Context, csr *certificatesv1.Certifica
 
 	return nil, nil
 }
+
+func (r *reconcileCSR) getHostEndpoint(ctx context.Context, hostname string) (*v3.HostEndpoint, error) {
+	hepList, err := r.calicoClient.ProjectcalicoV3().HostEndpoints().List(ctx, metav1.ListOptions{FieldSelector: fmt.Sprintf("spec.node=%s", hostname)})
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			return nil, nil
+		}
+		return nil, err
+	}
+	if len(hepList.Items) == 0 {
+		return nil, nil
+	}
+	return &hepList.Items[0], nil
```

Comment:
I suppose it would be possible to have more than 1 HEP for a given host.

Reply:
The HEP labels don't include a hostname by default. It is the user's resposibility to configure HEP labels.